packages/sourcecred/src/ui/components/LedgerViewer/LedgerEventRow.js

```diff
@@ -63,14 +63,21 @@ type LedgerEventRowProps = {|
 type EventRow = (LedgerEventRowProps) => ReactNode;
 
 function _friendlyFormatActionType(type: string): string {
-  //Split words by _, capitalize them, and join them with " "
-  var words = type.split("_");
-  words.forEach((word, idx) => {
-    word = word.toLowerCase();
-    word = word.charAt(0).toUpperCase() + word.slice(1);
-    words[idx] = word;
-  });
-  return words.join(" ");
+  //Memoize computed label values. Small set of expected labels,
+  //potentially large number of times displayed. Best to save the values.
+
+  const label = useMemo(() => {
+    //Split words by _, capitalize them, and join them with " "
+    var words = type.split("_");
+    words.forEach((word, idx) => {
+      word = word.toLowerCase();
+      word = word.charAt(0).toUpperCase() + word.slice(1);
```

Comment:
Almost there, use memo doesn't work outside a component this can be fixed by reverting the function like it was before and instead within the component at the top use it like this
```js
const type = props.event.action.type
const label = useMemo(() => _friendlyFormatActionType(type), [type])
```

and then evaluate the label inside JSX like this
```js
<Typography color="primary">
            {label}
 </Typography>
```